src/SprykerSdk/Acp/Configuration/SchemaGenerator/SchemaGenerator.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Acp\Configuration\SchemaGenerator;
+
+use cebe\openapi\spec\OpenApi;
+use SprykerSdk\Acp\Configuration\Reader\AppConfigurationReaderInterface;
+
+class SchemaGenerator implements SchemaGeneratorInterface
+{
+    protected AppConfigurationReaderInterface $appConfigurationReader;
+
+    protected SchemaWriterInterface $schemaWriter;
+
+    /**
+     * @param \SprykerSdk\Acp\Configuration\Reader\AppConfigurationReaderInterface $appConfigurationReader
+     * @param \SprykerSdk\Acp\Configuration\SchemaGenerator\SchemaWriterInterface $schemaWriter
+     */
+    public function __construct(AppConfigurationReaderInterface $appConfigurationReader, SchemaWriterInterface $schemaWriter)
+    {
+        $this->appConfigurationReader = $appConfigurationReader;
+        $this->schemaWriter = $schemaWriter;
+    }
+
+    /**
+     * @param string $configurationFilePath
+     * @param string $openapiFilePath
+     *
+     * @return bool
+     */
+    public function convertConfigurationToSchema(string $configurationFilePath, string $openapiFilePath): bool
+    {
+        $appConfigurationData = $this->appConfigurationReader->readConfigurationFile($configurationFilePath);
+
+        $generatedSchema = $this->generateSchemaFromConfiguration($appConfigurationData);
+
+        return $this->schemaWriter->writeSchemaToFile($generatedSchema, $openapiFilePath);
+    }
+
+    /**
+     * @param array $appConfigurationData
+     *
+     * @return \cebe\openapi\spec\OpenApi
+     */
+    protected function generateSchemaFromConfiguration(array $appConfigurationData): OpenApi
+    {
+        $appConfigurationSchema = $this->parseAppConfigurationData($appConfigurationData);
+
+        return new OpenApi([
+            'components' => [
+                'schemas' => [
+                    'AppConfiguration' => $appConfigurationSchema,
+                ],
+            ],
+        ]);
+    }
+
+    /**
+     * @param array $appConfigurationData
+     *
+     * @return array
+     */
+    protected function parseAppConfigurationData(array $appConfigurationData): array
```

Comment:
This looks like it belongs in it's own class.

Reply:
not sure - after removing writing part from SchemaGenerator (now SchemaConverter) it only does reading (by delegating request to Reader) and converting (this method)